src/compute/arithmetics/mod.rs

```diff
@@ -75,19 +75,13 @@ use super::arity::unary;
 // is an error then an ArrowError is return with the data_type that cause it.
 // It returns the result from the arithmetic_primitive function evaluated with
 // the Operator selected
-macro_rules! primitive_arithmetic_match {
-    ($lhs: expr, $rhs: expr, $op: expr, $primitive_array_type: ty) => {{
-        let res_lhs = $lhs
-            .as_any()
-            .downcast_ref::<$primitive_array_type>()
-            .unwrap();
-
-        let res_rhs = $rhs
-            .as_any()
-            .downcast_ref::<$primitive_array_type>()
-            .unwrap();
-
-        arithmetic_primitive(res_lhs, $op, res_rhs)
+macro_rules! primitive {
+    ($lhs: expr, $rhs: expr, $op: expr, $array_type: ty) => {{
+        let res_lhs = $lhs.as_any().downcast_ref().unwrap();
+
+        let res_rhs = $rhs.as_any().downcast_ref().unwrap();
+
+        arithmetic_primitive::<$array_type>(res_lhs, $op, res_rhs)
```

Comment:
You are always finding these little simplifications with dynamic allocators. Nice 😄